api/controllers/user.go

```diff
@@ -186,6 +186,16 @@ func (uc *UserController) FindByID(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 }
 
+// Delete user
+// @Summary Delete user
+// @Description Marks a user as deleted (soft delete) without removing them from the database
+// @Tags User
+// @Param id path int true "User ID"
+// @Success 204 "No Content"
+// @Failure 400 "Bad request"
+// @Failure 401 "Unauthorized"
+// @Failure 500 "Internal server error"
+// @Router /user/{userID} [delete]
 func (uc *UserController) Delete(w http.ResponseWriter, r *http.Request) {
 	userIDFromTokenStr := r.Header.Get("UserId")
```

Comment:
Essa rota tem dois parâmetros:

1) path
2) Header

O segundo não foi especificado na documentação swag, seria necessário colocar para que as pessoas que vão consumir a rota saibam da sua existência.

Reply:
O userId presente no header está em todas as rotas privadas e é responsável por garantir que o usuário esteja autenticado. Em determinadas rotas, como a de exclusão de usuários, o userId é verificado para assegurar que o usuário não está tentando excluir a conta de outra pessoa.

Esse userId é extraído do token JWT, que é obrigatório em todas as rotas privadas. O middleware de autenticação cuida dessa lógica: sempre que uma requisição é feita a uma rota privada, ela passa por essa verificação. Se o token for válido, o middleware adiciona o userId ao header da requisição.

![image](https://github.com/user-attachments/assets/e5496c94-90f5-42a5-8cf1-1891f86a617b)

Reply:
Excelente!